tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -13598,6 +13606,78 @@ INLINE void static add_property_struct(char *func_name, int nreductions,
 }
 #endif
 
+void 
+add_vscale() 
+{
+  char token[60];
+  sprintf(token, "%d,%d", flg.min, flg.max);
+  print_token(" \"vscale-range\"=\"");
+  print_token(token);
+  print_token("\"");
+}
+
+bool
+is_vscale_feat(char *mod)
+{
+  bool result = false;
+  int len = sizeof(*CPU_VSCALE)-1;
```

Comment:
that's not how sizeof should be used! `CPU_VSCALE` is an array of pointers, so a single element of it (`*CPU_VSCALE`) is a pointer, on 64-bit machine it is 8 bytes in size, so sizeof returns always 8 here. You were lucky that `CPU_VSCALE` contains 7 elements and that `8 - 1` is also 7.